azimuth_capi/status.py

```diff
@@ -142,6 +143,23 @@ def _reconcile_cluster_phase(cluster):
         cluster.status.phase = ClusterPhase.UNHEALTHY
     else:
         cluster.status.phase = ClusterPhase.READY
+        # reset the timeout timestamp, as we are now in a stable state
+        cluster.status.last_updated_timestamp = None
+
+    # If we hit a terminal state, remove the updated timestamp,
+    if cluster.status.phase in {ClusterPhase.READY, ClusterPhase.FAILED}:
+        cluster.status.last_updated_timestamp = None
+    else:
+        # if not a terminal state, ensure timestamp has been set
+        if cluster.status.last_updated_timestamp is None:
+            cluster.status.last_updated_timestamp = dt.datetime.now(dt.timezone.utc)
+
+    # timeout pending states if we are stuck their too long
+    if cluster.status.phase in {ClusterPhase.PENDING, ClusterPhase.RECONCILING, ClusterPhase.UPGRADING}:
+        now = dt.datetime.now(dt.timezone.utc)
+        timeout_after = cluster.status.last_updated_timestamp + dt.timedelta(minutes=settings.cluster_timeout_minutes)
```

Comment:
Couple of very long lines here. I know we don't have a linter enabled on this repo, but I try to stick to 100 columns if possible.